### PR TITLE
Kinetica exporter/pr 1

### PR DESCRIPTION
**Pull request by am-kinetica**
**Description:** First PR for the Kinetica exporter
Contains what has been mandated by the document - https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md
**Link to tracking Issue:** (https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24294)

**Testing:** Factory and config tests have been added

**Documentation:** Readme.md has been provided

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
@am-kinetica please sign the CLA

File: exporter/kineticaexporter/README.md
Comment (TylerHelmuth) [inline exporter/kineticaexporter/README.md:1]: Please run `make generate` to add the OpenTelemetry header to the top of this file

File: exporter/kineticaexporter/config.go
Comment (TylerHelmuth) [inline exporter/kineticaexporter/config.go:24]: When you run `make generate` it will generate a type string for you to use based on the metadata.yaml file.

File: exporter/kineticaexporter/config.go
Comment (TylerHelmuth) [inline exporter/kineticaexporter/config.go:82]: Is this part of the exporting logic? If it is creating a logger to use within the component but not as part of the data exporting then we should be using the logger that the collector provides instead.  If it is exporting logic please remove it from this PR and we'll add it in a future PR.

File: exporter/kineticaexporter/config.go
Comment (TylerHelmuth) [inline exporter/kineticaexporter/config.go:165]: Is this part of the exporting logic? If it is creating a logger to use within the component but not as part of the data exporting then we should be using the logger that the collector provides instead. If it is exporting logic please remove it from this PR and we'll add it in a future PR.

File: exporter/kineticaexporter/config_log_zap.yaml
Comment (TylerHelmuth) [inline exporter/kineticaexporter/config_log_zap.yaml:1]: Is this extra configuration for exporting or logging?  If logging we should be using the collector's logger in all instances.  If exporting then please remove from this PR and we'll add back later.

File: exporter/kineticaexporter/factory.go
Comment (TylerHelmuth) [inline exporter/kineticaexporter/factory.go:36]: Are these meaningful defaults? Could the exporter connect to a database running at `http://localhost:9191/` with these credentials?

File: exporter/kineticaexporter/factory.go
Comment (TylerHelmuth) [inline exporter/kineticaexporter/factory.go:33]: Is it a common use case to run the database locally?  Would it be better to default this to `""` and validate that a value is supplied by the user?

File: exporter/kineticaexporter/logs_exporter.go
Comment (TylerHelmuth) [inline exporter/kineticaexporter/logs_exporter.go:26]: We shouldn't need to call validate ourselves, the collector's framework will handle that for us. This applies to all signals.

File: exporter/kineticaexporter/metadata.yaml
Comment (TylerHelmuth) [inline exporter/kineticaexporter/metadata.yaml:7]:
```suggestion
  distributions: []
```

This is not included in the official Contrib distribution yet

File: cmd/otelcontribcol/builder-config.yaml
Comment (TylerHelmuth) [inline cmd/otelcontribcol/builder-config.yaml:63]: Lets modify this command during a later PR.  The component is not functional in its current state so it doesn't need to be added to this build manifest yet.

File: exporter/kineticaexporter/metrics_exporter.go
Comment (TylerHelmuth) [inline exporter/kineticaexporter/metrics_exporter.go:48]: For all signals, please remove all business logic around exporting.  This PR should only be the factory and config. You'll need these functions to satisfy the factory, but for now it only return nil.

File: exporter/kineticaexporter/sql/datamodel.sql
Comment (TylerHelmuth) [inline exporter/kineticaexporter/sql/datamodel.sql:1]: What is the purpose of this file?

Comment (am-kinetica) [reply exporter/kineticaexporter/sql/datamodel.sql:1]: This doesn't play any role in exporting as such. We have just kept it for reference in case someone wants to pre-create the tables before running the exporter. If we want to provide such a script what would be an appropriate place for that ?

File: exporter/kineticaexporter/README.md
Comment (am-kinetica) [reply exporter/kineticaexporter/README.md:1]: I tried running `make generate` after removing everything I had in our `readme.md` but it generates nothing; the file is empty.

File: exporter/kineticaexporter/config.go
Comment (am-kinetica) [reply exporter/kineticaexporter/config.go:82]: Removed this method since this was intended to be specific to our internal logging requirements.

File: exporter/kineticaexporter/config.go
Comment (am-kinetica) [reply exporter/kineticaexporter/config.go:165]: Removed this method since this was intended to be specific to our internal logging requirements.

File: exporter/kineticaexporter/config_log_zap.yaml
Comment (am-kinetica) [reply exporter/kineticaexporter/config_log_zap.yaml:1]: Removed this for the time being, to be added back in a future PR.

File: exporter/kineticaexporter/factory.go
Comment (am-kinetica) [reply exporter/kineticaexporter/factory.go:36]: Replaced them with what we consider as meaningful defaults which are empty strings for both values.

File: exporter/kineticaexporter/factory.go
Comment (am-kinetica) [reply exporter/kineticaexporter/factory.go:33]: There are cases where the database could be local; so we think we could keep this as a valid default.

File: exporter/kineticaexporter/logs_exporter.go
Comment (am-kinetica) [reply exporter/kineticaexporter/logs_exporter.go:26]: Removed this check from all signals.

File: exporter/kineticaexporter/metadata.yaml
Comment (am-kinetica) [reply exporter/kineticaexporter/metadata.yaml:7]: Modified as suggested

File: exporter/kineticaexporter/metrics_exporter.go
Comment (am-kinetica) [reply exporter/kineticaexporter/metrics_exporter.go:48]: Removed all business logic as suggested and the methods return `nil` for now.

File: exporter/kineticaexporter/README.md
Comment (TylerHelmuth) [reply exporter/kineticaexporter/README.md:1]: Sorry, I forgot you need to add these lines to the top of the file.

```
<!-- status autogenerated section -->
<!-- end autogenerated section -->
```

Comment (am-kinetica) [reply exporter/kineticaexporter/README.md:1]: Tried this, still no luck.

Comment (TylerHelmuth) [reply exporter/kineticaexporter/README.md:1]: Can you tell it has been a while since I worked on a component from scratch? haha I forgot you also need this like in your doc.go file: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b0056fae918781e0daabd1cb76b4050d38d76a9e/processor/transformprocessor/doc.go#L4

File: cmd/otelcontribcol/go.mod
Comment (TylerHelmuth) [inline cmd/otelcontribcol/go.mod:1]: We can remove any changes to this file.  We'll add the component to this local build once we add the business logic in a future PR.

File: exporter/kineticaexporter/sql/datamodel.sql
Comment (TylerHelmuth) [inline exporter/kineticaexporter/sql/datamodel.sql:1]: We should remove this file as it is unneeded for the operation of the component.  Its contents would better be housed in your docs site.

**Review by TylerHelmuth — COMMENTED** [review]
@am-kinetica things are lookin good for this PR.  Left a couple more thoughts, and we'll need the CLA signed before we can merge.

File: exporter/kineticaexporter/README.md
Comment (am-kinetica) [reply exporter/kineticaexporter/README.md:1]: It's still the same. Now the file has:
`// Copyright The OpenTelemetry Authors
// SPDX-License-Identifier: Apache-2.0

//go:generate mdatagen metadata.yaml
`
Nothing is being generated

File: cmd/otelcontribcol/builder-config.yaml
Comment (am-kinetica) [reply cmd/otelcontribcol/builder-config.yaml]: This line has been removed.

File: exporter/kineticaexporter/sql/datamodel.sql
Comment (am-kinetica) [reply exporter/kineticaexporter/sql/datamodel.sql:1]: Removed this file and put it in a section in the `README.md`.

File: exporter/kineticaexporter/config.go
Comment (am-kinetica) [reply exporter/kineticaexporter/config.go:43]: This was a test code which has been removed now.

File: exporter/kineticaexporter/README.md
Comment (am-kinetica) [reply exporter/kineticaexporter/README.md:1]: Added the `doc.go` file and re-generated; this time the header section has been generated.

File: exporter/kineticaexporter/config.go
Comment (TylerHelmuth) [reply exporter/kineticaexporter/config.go:16]: @am-kinetica This is the last thing to update.  We can remove this constant and use the value from `internal/metadata`

File: cmd/otelcontribcol/go.mod
Comment (am-kinetica) [reply cmd/otelcontribcol/go.mod]: Removed this line.

File: exporter/kineticaexporter/config.go
Comment (am-kinetica) [reply exporter/kineticaexporter/config.go]: Removed this with the generated value

File: exporter/kineticaexporter/factory.go
Comment (TylerHelmuth) [inline exporter/kineticaexporter/factory.go:4]:
```suggestion
```

We should only need this line in `doc.go`.

Comment (am-kinetica) [reply exporter/kineticaexporter/factory.go]: Removed this line.

**Review by TylerHelmuth — APPROVED** [review]
LGTM

**Comment by TylerHelmuth** [comment]
@atoulme can you help out with the failing check-codeowners? I though being added to the allowlist was enough

**Comment by atoulme** [comment]
You still need to edit `.github/CODEOWNERS` and add the path with the codeowners.

**Comment by TylerHelmuth** [comment]
@atoulme both @am-kinetica and I are getting access issues when running `make update-codeowners`.  When I create a PAT and set the `GITHUB_TOKEN` env var I get nil pointer reference errors.

**Comment by TylerHelmuth** [comment]
@am-kinetica in the meantime you can manually update the CODEOWNERS file with your new exporter:

`exporter/kineticaexporter/   @open-telemetry/collector-contrib-approvers @am-kinetica @TylerHelmuth`

**Comment by TylerHelmuth** [comment]
@am-kinetica if you only accepted my suggestion I believe you'll need a `go mod tidy` to go along with it.

**Comment by TylerHelmuth** [comment]
@atoulme we're seeing an issue with checkapi as well. What qualifies for being in the allowlist?

**Pull request merged**
